lib/cli/src/cli.rs

```diff
@@ -344,14 +344,34 @@ fn test_split_version() {
     );
 }
 
+#[derive(Debug)]
+pub(crate) enum SplitVersionError {
+    InvalidCommandName(anyhow::Error, Option<anyhow::Error>),
```

Comment:
Apart from the other comment, a `(Error, Option<Error>)` tuple is not descriptive at all.
I'd use a struct variant with more descriptive names for the fields.